frontend/packages/ux-editor/src/components/config/editModal/EditDataModelBinding/EditDataModelBinding.tsx

```diff
@@ -0,0 +1,67 @@
+import type { IGenericEditComponent } from '../../componentConfig';
+import type { ComponentType } from 'app-shared/types/ComponentType';
+import React, { useState } from 'react';
+import classes from './EditDataModelBinding.module.css';
+import { useTranslation } from 'react-i18next';
+import { UndefinedBinding } from './UndefinedBinding';
+import { EditBinding } from './EditBinding';
+import { DefinedBinding } from './DefinedBinding';
+import { convertDataBindingToInternalFormat } from '../../../../utils/dataModelUtils';
+
+export interface EditDataModelBindingProps<T extends ComponentType>
+  extends IGenericEditComponent<T> {
+  renderOptions?: {
+    label?: string;
+    returnValue?: any;
+    key?: string;
+    uniqueKey?: any;
+  };
+  helpText?: string;
+}
+
+export const EditDataModelBinding = <T extends ComponentType>({
+  component,
+  handleComponentChange,
+  renderOptions,
+  helpText,
+}: EditDataModelBindingProps<T>) => {
+  const { uniqueKey, key, label } = renderOptions || {};
+  const bindingKey = key || 'simpleBinding';
+  const { t } = useTranslation();
+  const [dataModelSelectVisible, setDataModelSelectVisible] = useState(false);
+
+  const internalBindingFormat = convertDataBindingToInternalFormat(component, bindingKey);
+
+  const labelSpecificText = label
+    ? t(`ux_editor.modal_properties_data_model_label.${label}`)
+    : t(`ux_editor.component_title.${component.type}`);
+
+  return (
+    <div key={uniqueKey || ''} className={classes.wrapper}>
```

Comment:
Probably only a refactoring that existed from before; but what is the uniqueKey compared to the key? Could we use the key and delete uniqueKey? 🤔

Reply:
I have no idea ^^ I deleted the uniqueKey